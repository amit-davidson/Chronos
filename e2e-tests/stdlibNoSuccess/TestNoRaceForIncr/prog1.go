package main

func main() {
	done := make(chan bool)
	x := 0
	go func() {
		x++
		done <- true
	}()
	for i := 0; i < 0; x++ {
	}
	<-done
}
