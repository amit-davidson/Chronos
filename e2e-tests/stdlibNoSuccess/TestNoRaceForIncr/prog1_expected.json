{
	"Lockset": {
		"ExistingLocks": {},
		"ExistingUnlocks": {}
	},
	"GuardedAccess": [
		{
			"ID": 1,
			"PosID": 1,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:4:2",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:4:14",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 1
				}
			}
		},
		{
			"ID": 2,
			"PosID": 2,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:4:2",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:4:2",
			"OpKind": 1,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 2
				}
			}
		},
		{
			"ID": 3,
			"PosID": 3,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:5:2",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 3
				}
			}
		},
		{
			"ID": 4,
			"PosID": 4,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:5:2",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:5:2",
			"OpKind": 1,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 4
				}
			}
		},
		{
			"ID": 5,
			"PosID": 5,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:7:3",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:5:2",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 2,
				"Clock": {
					"1": 5,
					"2": 1
				}
			}
		},
		{
			"ID": 6,
			"PosID": 6,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:7:3",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:7:3",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 2,
				"Clock": {
					"1": 5,
					"2": 2
				}
			}
		},
		{
			"ID": 7,
			"PosID": 7,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:7:3",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 2,
				"Clock": {
					"1": 5,
					"2": 3
				}
			}
		},
		{
			"ID": 8,
			"PosID": 8,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:7:3",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:7:3",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 2,
				"Clock": {
					"1": 5,
					"2": 4
				}
			}
		},
		{
			"ID": 9,
			"PosID": 9,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:7:3",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:5:2",
			"OpKind": 1,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 2,
				"Clock": {
					"1": 5,
					"2": 5
				}
			}
		},
		{
			"ID": 10,
			"PosID": 10,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:8:3",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:4:2",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 2,
				"Clock": {
					"1": 5,
					"2": 6
				}
			}
		},
		{
			"ID": 11,
			"PosID": 11,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:10:21",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:5:2",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 6
				}
			}
		},
		{
			"ID": 12,
			"PosID": 12,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:10:21",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:10:21",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 7
				}
			}
		},
		{
			"ID": 13,
			"PosID": 13,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:10:21",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 8
				}
			}
		},
		{
			"ID": 14,
			"PosID": 14,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:10:21",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:10:21",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 9
				}
			}
		},
		{
			"ID": 15,
			"PosID": 15,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:10:21",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:5:2",
			"OpKind": 1,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 10
				}
			}
		},
		{
			"ID": 16,
			"PosID": 16,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:12:4",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:4:2",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 11
				}
			}
		},
		{
			"ID": 17,
			"PosID": 17,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:12:2",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:12:4",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 12
				}
			}
		},
		{
			"ID": 18,
			"PosID": 18,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:10:16",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 13
				}
			}
		},
		{
			"ID": 19,
			"PosID": 19,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:10:16",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 14
				}
			}
		},
		{
			"ID": 20,
			"PosID": 20,
			"Pos": "",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlibNoSuccess/TestNoRaceForIncr/prog1.go:10:16",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 15
				}
			}
		}
	]
}